### PR TITLE
feat: Integrate one click binding to sourceData of select and multi select widget

### PR DESCRIPTION
**Pull request by sbalaji1192**
## Description
We have changed the property control of sourceData of select and multi select widget to dropdowns which has one click-binding items.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/24780

#### Type of change

- New feature (non-breaking change which adds functionality)

- [x] Manual
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [x] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by laveena-en** [comment]
@sbalaji1192 

Connect data option must be disabled until label and value options are chosen: https://www.loom.com/share/e22efff0a20b4b6b891d308ed47218cd

Unable to switch to sample data on selecting any DS first. Works fine on refresh: https://www.loom.com/share/02a5c24d1a424a1297ed234cafffe83e?sid=9a883962-ab9e-46c9-9c7d-6305081e3ffb

Formatting of data is skewed on switching between sample data and any DS: https://www.loom.com/share/afa7bb2befd24e59aaf313fe82e5a46d

Error is seen on trying to bind a created query with an existent widget: https://www.loom.com/share/c38e11889aa74478823219c128c58955

^ Similarly, on trying to bind the query with a new widget, the configuration isn’t as expected: https://www.loom.com/share/80e14f57dc6d46ee9b110198c5310dbc

Can we hide sample data's properties when a datasource is selected instead: https://www.loom.com/share/66ae89fa97d84c0f9310f8808f9429ab

**Comment by sbalaji1192** [comment]
@laveena-en 
> Error is seen on trying to bind a created query with an existent widget:
Expected. you need to select the label and value after binding a new query to select widget

>Can we hide sample data's properties when a data source is selected instead:
we will need design input. Let's create a separate issue for this.

Fixed the rest of the callouts.

**Comment by laveena-en** [comment]
Tested and verified this PR. LGTM
Test plan for coverage: https://github.com/appsmithorg/appsmith/pull/25750

Raised a separate issue for the last callout that needed design input [here](https://github.com/appsmithorg/appsmith/issues/26122)

Issues with suggested widgets requires BE's intervention. Discussion [here](https://theappsmith.slack.com/archives/C040LHZN03V/p1691401670836309)

File: app/client/src/components/editorComponents/WidgetQueryGeneratorForm/WidgetSpecificControls/index.tsx [hunk 1]
```diff
@@ -1,37 +1,44 @@
...
-    aliasPicker = props.aliases.map((alias) => {
-      <ColumnDropdown alias={`alias.${alias}`} label={alias} onSelect={noop} />;
-    });
+  if (props.aliases?.length) {
+    aliases = props.aliases.map(({ name }) => (
+      <ColumnDropdown
+        alias={`alias.${name}`}
+        id={name}
+        key={name}
+        label={name.slice(0, 1).toUpperCase() + name.slice(1)}
```
Comment (berzerkeer) [inline app/client/src/components/editorComponents/WidgetQueryGeneratorForm/WidgetSpecificControls/index.tsx:32]: nit : would be better if we can move this expression to its own vairable. 

const label = name.slice(0, 1).toUpperCase() + name.slice(1)

File: app/client/src/WidgetQueryGenerators/GSheets/index.ts [hunk 2]
```diff
@@ -54,54 +54,67 @@ export default abstract class GSheets extends BaseQueryGenerator {
...
             formConfig.tableHeaderIndex,
           ),
         },
-        dynamicBindingPathList: [
-          {
-            key: "formData.where.data",
-          },
-          {
-            key: "formData.sortBy.data",
+        dynamicBindingPathList: [],
```
Comment (berzerkeer) [inline app/client/src/WidgetQueryGenerators/GSheets/index.ts:68]: Curious , why have we removed these ?

**Review by berzerkeer — APPROVED** [review]

File: app/client/src/WidgetQueryGenerators/GSheets/index.ts (hunk 2 above)
Comment (sbalaji1192) [reply app/client/src/WidgetQueryGenerators/GSheets/index.ts:68]: We have added these items conditionally below

**Review by berzerkeer — APPROVED** [review]

**Pull request merged**